app/services/move_project_to_channel.rb

```diff
@@ -1,13 +1,21 @@
 class MoveProjectToChannel
-  def initialize(project_params,channel_params)
-    @project = Project.find(project_params)
-    @channel = Channel.find(channel_params)
+  def initialize(project, channel)
+    @project = project
+    @channel = channel
   end
 
-  def call
-    if project.channels.empty?
-      project.channels << channel
+  def self.perform(project, channel)
+    new(project, channel).perform
+  end
+
+  def perform
+    if project.present? && channel.present? && project.able_to_move_to_channel?
```

Comment:
Doing what I suggested on the controller, you don't need to check if both project and channel are present.